extensions/gitpod/src/remoteConnector.ts

```diff
@@ -403,40 +481,170 @@ export default class LocalApp extends Disposable {
+		}
+
+		const currentConfigFile = remoteSSHconfig.get<string>('configFile');
+		if (usingSSHGateway) {
+			if (currentConfigFile?.includes('gitpod_ssh_config')) {
+				await remoteSSHconfig.update('configFile', undefined, vscode.ConfigurationTarget.Global);
+			}
+		} else {
+			// TODO(ak) notify a user about config file changes?
+			if (currentConfigFile === localAppSSHConfigPath) {
+				// invalidate cached SSH targets from the current config file
+				await remoteSSHconfig.update('configFile', undefined, vscode.ConfigurationTarget.Global);
+			}
+			await remoteSSHconfig.update('configFile', localAppSSHConfigPath, vscode.ConfigurationTarget.Global);
+		}
+	}
+
+	private async ensureRemoteSSHExtInstalled(): Promise<boolean> {
+		const msVscodeRemoteExt = vscode.extensions.getExtension('ms-vscode-remote.remote-ssh');
+		if (msVscodeRemoteExt) {
+			return true;
+		}
+
+		const install = 'Install';
+		const cancel = 'Cancel';
+		const action = await vscode.window.showInformationMessage('Please install "Remote - SSH" extension to connect to a Gitpod workspace.', install, cancel);
+		if (action === cancel) {
+			return false;
+		}
+
+		this.logger.info('Installing "ms-vscode-remote.remote-ssh" extension');
+
+		await vscode.commands.executeCommand('extension.open', 'ms-vscode-remote.remote-ssh');
+		await vscode.commands.executeCommand('workbench.extensions.installExtension', 'ms-vscode-remote.remote-ssh');
+
+		return true;
+	}
+
+	public async handleUri(uri: vscode.Uri) {
+		if (uri.path === RemoteConnector.AUTH_COMPLETE_PATH) {
+			this.logger.info('auth completed');
+			return;
+		}
+
+		const isRemoteSSHExtInstalled = this.ensureRemoteSSHExtInstalled();
+		if (!isRemoteSSHExtInstalled) {
+			return;
+		}
+
+		const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+		const forceUseLocalApp = vscode.workspace.getConfiguration('gitpod').get<boolean>('remote.useLocalApp')!;
+
+		const params: SSHConnectionParams = JSON.parse(uri.query);
+		if (new URL(params.gitpodHost).host !== new URL(gitpodHost).host) {
+			const yes = 'Yes';
+			const cancel = 'Cancel';
+			const action = await vscode.window.showInformationMessage(`Trying to connect to a remote workspace in a different Gitpod Host. Continue and update 'gitpod.host' setting to '${params.gitpodHost}'?`, yes, cancel);
+			if (action === cancel) {
+				return;
+			}
+
+			await vscode.workspace.getConfiguration('gitpod').update('host', params.gitpodHost, vscode.ConfigurationTarget.Global);
+			this.logger.info(`Updated 'gitpod.host' setting to '${params.gitpodHost}' while trying to connect to a remote workspace`);
+		}
+
+		this.logger.info('Opening remote workspace', uri.toString());
+
+		let sshDestination: string | undefined;
+		if (!forceUseLocalApp) {
+			try {
+				this.telemetry.sendTelemetryEvent('gitpod_desktop_ssh', { kind: 'gateway', status: 'connecting' });
```

Comment:
`gitpod_desktop_ssh` it is bad name, we have otehr desktop ides, it is better to be clear that it is `vscode_desktop_ssh`. gitpod is obvious.

Reply:
Should I update the other events that have `gitpod_` prefix or only this new events will have `vscode_` prefix?

Reply:
No, let's use for others as is now already. But in future let's always give very specific names.